lib/galaxy/webapps/galaxy/api/item_tags.py

```diff
@@ -3,71 +3,93 @@
-    tagged_item_id = "workflow_id"
+router = Router(tags=["item_tags"])
+
+
+@router.cbv
+class FastAPIItemTags:
+    manager: ItemTagsManager = depends(ItemTagsManager)
+
+    @classmethod
+    def create_class(cls, prefix, tagged_item_class):
+        class Temp(cls):
+            @router.get(f"/api/{prefix}/{{tagged_item_id}}/tags")
+            def index(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                tagged_item_id: DecodedDatabaseIdField = Path(..., title="Item ID"),
+            ) -> ItemTagsListResponse:
+                return self.manager.index(trans, tagged_item_class, tagged_item_id)
+
+            @router.get(f"/api/{prefix}/{{tagged_item_id}}/tags/{{tag_name}}")
+            def show(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                tagged_item_id: DecodedDatabaseIdField = Path(..., title="Item ID"),
+                tag_name: str = Path(..., title="Tag Name"),
+            ) -> ItemTagsResponse:
+                return self.manager.show(trans, tagged_item_class, tagged_item_id, tag_name)
+
+            @router.post(f"/api/{prefix}/{{tagged_item_id}}/tags/{{tag_name}}")
+            def create(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                tagged_item_id: DecodedDatabaseIdField = Path(..., title="Item ID"),
+                tag_name: str = Path(..., title="Tag Name"),
+                payload: ItemTagsCreatePayload = Body(...),
+            ) -> ItemTagsResponse:
+                return self.manager.create(trans, tagged_item_class, tagged_item_id, tag_name, payload)
+
+            @router.put(f"/api/{prefix}/{{tagged_item_id}}/tags/{{tag_name}}")
+            def update(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                tagged_item_id: DecodedDatabaseIdField = Path(..., title="Item ID"),
+                tag_name: str = Path(..., title="Tag Name"),
+                payload: ItemTagsCreatePayload = Body(...),
+            ) -> ItemTagsResponse:
+                return self.manager.create(trans, tagged_item_class, tagged_item_id, tag_name, payload)
+
+            @router.delete(f"/api/{prefix}/{{tagged_item_id}}/tags/{{tag_name}}")
+            def delete(
+                self,
+                trans: ProvidesAppContext = DependsOnTrans,
+                tagged_item_id: DecodedDatabaseIdField = Path(..., title="Item ID"),
+                tag_name: str = Path(..., title="Tag Name"),
+            ) -> bool:
+                return self.manager.delete(trans, tagged_item_class, tagged_item_id, tag_name)
+
+        return Temp
+
+
+prefixs = {
+    "histories": "History",
+    "histories/{history_id}/contents": "HistoryDatasetAssociation",
+    "workflows": "StoredWorkflow",
+}
+for prefix, tagged_item_class in prefixs.items():
+    router.cbv(FastAPIItemTags.create_class(prefix, tagged_item_class))
```

Comment:
That's a clever approach, I like it a lot! It's a little unfortunate that this shows up in the API schema as `tagged_item_id`, do you see a way to adapt this so that it would be a more meaningful variable ? It's not worse than before though, so definitely very cool.

Reply:
Thank you! I changed `tagged_item_id` to `id` and added a description to show it in docs.